src/librustdoc/html/layout.rs

```diff
@@ -98,7 +98,7 @@ crate fn render<T: Print, S: Print>(
                            placeholder=\"Click or press ‘S’ to search, ‘?’ for more options…\" \
                            type=\"search\">\
                 </div>\
-                <button class=\"help-button\">?</button>
+                <a href=\"javascript:;\" class=\"help-button\">?</a>
```

Comment:
I just experimented a bit and you can keep the button. All you need to do is add `type="button"` and it works:

```suggestion
                <button type="button" class=\"help-button\">?</button>
```

Reply:
> In HTML, when a button follows an input, if the enter keep is pressed on the input, instead of sending the submit event to the input, it'll create a click event on the button following it

I don't think that is what's happening here. Based on experimentation, reading the output of rustdoc, and looking at MDN it seems that this is the issue:

1. The button is inside a `<form>`:

   ```html
   <form class="search-form">
     <div class="search-container">
       <div>
         <select id="crate-search">
           <option value="All crates">
             All crates
           </option>
         </select><input class="search-input" name="search"
         autocomplete="off" spellcheck="false" placeholder=
         "Click or press &acirc;&euro;&tilde;S&acirc;&euro;&trade; to search, &acirc;&euro;&tilde;?&acirc;&euro;&trade; for more options&acirc;&euro;&brvbar;"
         type="search">
       </div><button class="help-button">?</button> <a id=
       "settings-menu" href="../settings.html" name=
       "settings-menu"><img src="../wheel.svg" alt="Change settings"
       width="18"></a>
     </div>
   </form>
   ```

2. [According to MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Attributes), the default button behavior will be `submit` if the button is inside a `<form>` and `type` is not specified:

   > `submit`: The button submits the form data to the server. This is the default if the attribute is not specified for buttons associated with a `<form>`, or if the attribute is an empty or invalid value.

So just adding `type="button"` solves the problem.

Reply:
Yes it was one of the two solutions I came across. I just feel "less good" about this one. No strong feeling though, it just felt a "more obvious" to instead switch to a fake link.

Reply:
Hmm, I think a *fake* link is worse ;)

Reply:
Based on the thumbs-ups I've been getting, I think Joshua agrees with me.

Since you don't have strong feelings, and we both prefer the button, how about we go with the button? (I think it's likely more accessible to have a button too because it's more 'semantically-correct'.)

Reply:
Yes, I agree :laughing:

Reply:
Then let's go for a button. :)